scripts/synthetix/facilitate_trade.js

```diff
@@ -73,44 +98,75 @@ module.exports = async (callback) => {
     const formatedRate = snxjs.utils.formatEther(exchangeRate)
-    console.log("sETH Price", snxjs.utils.formatEther(exchangeRate))
+    console.log("Oracle sETH Price (in sUSD)", formatedRate)
+
+    const exchangeBuyETHPrice = await priceEstimation(
+      sETH.exchangeId,
+      sUSD.exchangeId,
+      toWEI(MIN_SELL_USD, sUSD.decimals),
+      networkId
+    )
+    console.log("Exchange buy  sETH price (in sUSD)", 1 / exchangeBuyETHPrice)
+
+    const minSellETH = toWEI(MIN_SELL_USD / formatedRate, sETH.decimals)
+    const exchangeSellETHPrice = await priceEstimation(sUSD.exchangeId, sETH.exchangeId, minSellETH, networkId)
+    console.log("Exchange sell sETH price (in sUSD)", exchangeSellETHPrice)
 
     // Using synthetix's fees, and formatting their return values with their tools, plus parseFloat.
     const sETHTosUSDFee = parseFloat(snxjs.utils.formatEther(await snxjs.Exchanger.feeRateForExchange(sETHKey, sUSDKey)))
     const sUSDTosETHFee = parseFloat(snxjs.utils.formatEther(await snxjs.Exchanger.feeRateForExchange(sUSDKey, sETHKey)))
 
-    // Compute buy-sell amounts based on unlimited orders with rates from above.
-    const { base: sellSUSDAmount, quote: buyETHAmount } = getUnlimitedOrderAmounts(
-      1 / (formatedRate * (1 - sUSDTosETHFee)),
-      sETH.decimals,
-      sUSD.decimals
-    )
-    const { base: sellETHAmount, quote: buySUSDAmount } = getUnlimitedOrderAmounts(
-      formatedRate * (1 + sETHTosUSDFee),
-      sUSD.decimals,
-      sETH.decimals
-    )
-
-    const buyAmounts = [buyETHAmount, buySUSDAmount]
-    const sellAmounts = [sellSUSDAmount, sellETHAmount]
-
-    // Fetch auction index and declare validity interval for orders.
-    // Note that order validity interval is inclusive on both sides.
-    const batchId = (await exchange.getCurrentBatchId.call()).toNumber()
-    const validFroms = Array(2).fill(batchId)
-    const validTos = Array(2).fill(batchId)
-
-    // Avoid querying exchange by tokenAddress for fixed tokenId
-    const buyTokens = [sETH, sUSD].map((token) => token.exchangeId)
-    const sellTokens = [sUSD, sETH].map((token) => token.exchangeId)
-
-    const gasPrices = await (await fetch(gasStationURL[networkId])).json()
-    const scaledGasPrice = parseInt(gasPrices[argv.gasPrice] * argv.gasPriceScale)
-    console.log(`Using current "${argv.gasPrice}" gas price scaled by ${argv.gasPriceScale}: ${scaledGasPrice}`)
-    await exchange.placeValidFromOrders(buyTokens, sellTokens, validFroms, validTos, buyAmounts, sellAmounts, {
-      from: account,
-      gasPrice: scaledGasPrice,
-    })
+    // Initialize all arrays needed for potential order placement.
+    const [buyAmounts, sellAmounts, buyTokens, sellTokens] = [[], [], [], []]
+
+    // Compute buy-sell amounts based on unlimited orders with rates from above when the price is right.
+    const ourBuyETHRate = formatedRate * (1 - sUSDTosETHFee)
+    if (ourBuyETHRate < exchangeBuyETHPrice) {
```

Comment:
Shouldn't you be comparing the buy amount of the exchange with our *sell* rate instead? "They are willing to buy at a price higher than we are willing to sell" is what I'd expect here. Same below.

Reply:
I found this very confusing at the time of writing as well. If the exchange is offering a price lower than what we are willing to pay, then we place an order.

 I must admit that I made the  inequalities based on the fact that there were no orders being placed at the time of coding. I also wrote the logs in a way that would make me confident in the inequalities as they are:

```
Oracle sETH Price (in sUSD) 243.3257626212688802
Exchange buy  sETH price (in sUSD) 246.6144973941891
Exchange sell sETH price (in sUSD) 233.72582778494302
Not placing buy  sETH order, our rate of 242.60 is too low  for exchange.
Not placing sell sETH order, our rate of 244.06 is too high for exchange.
```

I will double check this, but at the time of writing I was pretty sure I had the inequalities the correct way around.

Reply:
I do agree this inequality looks completely backwards though.

Reply:
Oh I see what you mean. We should be comparing our buy price with their sell price and vice versa.

Reply:
Ok, so I haven't exactly got everything worked out here yet, but I would like to clarify that exchange price values here are the prices at which a user to realize. This can be seen by the fact that `exchangeBuyPrice > exchangeSellPrice` when there are no overlapping orders. That is, if you are buying ETH you can expect to pay `exchangeBuyPrice` and vice versa. So, although the values might appear confusing at first, or possibly the semantics are unclear, I think the values being compared make sense... Whether the inequalities are backwards is a different story I need to look into closer.

Reply:
Turns out the comparison variables were correctly named, but this particular line in the code was missing a multiplicative inversion...